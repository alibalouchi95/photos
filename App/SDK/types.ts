import { AppStateStatus } from 'react-native'
import { NodeState } from '../Models/TextileTypes'
export interface DiscoveredCafe {
  readonly peer: string
  readonly address: string
  readonly api: string
  readonly protocol: string
  readonly node: string
  readonly url: string
}
export interface DiscoveredCafes {
  readonly primary: DiscoveredCafe
  readonly secondary: DiscoveredCafe
}

export interface TextileOptions {
  debug?: boolean
}
export interface StoredNodeState {
  state: NodeState
  error?: string
}
export type TextileAppStateStatus = AppStateStatus | 'unknown' | 'backgroundFromForeground'
