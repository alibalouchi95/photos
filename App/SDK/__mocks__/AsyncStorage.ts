export default class AsyncStorage {
  setItem = jest.fn((key: string, value: any) => {
      return new Promise((resolve) => {
        this.mockSet(key, value)
        resolve()
      })
  })
  getItem = jest.fn((key: string) => {
      return new Promise((resolve) => {
          resolve(this.mockGet(key))
      })
  })

  multiRemove = jest.fn(() => {
    return new Promise((resolve) => {
      this.data = {}
      resolve()
    })
  })

  data = {}
  mockSet = (key: string, value: any) => {
    this.data[key] = value
  }
  mockGet = (key) => {
    return this.data[key]
  }
}
