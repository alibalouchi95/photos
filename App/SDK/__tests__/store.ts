import AsyncStorage from '../__mocks__/AsyncStorage'
import TextileStore from '../store'
import { NodeState, StoredNodeState } from '../types'

const store = new TextileStore()

const storedNodeState: StoredNodeState = {
  state: NodeState.nonexistent
}

jest.mock('react-native', () => {
  return {
      AsyncStorage
  }
})

describe('rn store', () => {
  describe('defaults', () => {
    it('serialize in and out to same', async () => {
      const serializedNodeState = store.serialize(storedNodeState) as string
      await expect(JSON.parse(serializedNodeState) as StoredNodeState).toEqual(storedNodeState)
    })

    it('empty state', async () => {
      await expect(store.getAppState()).resolves.toMatchSnapshot()
      await expect(store.getNodeOnline()).resolves.toMatchSnapshot()
      await expect(store.getNodeState()).resolves.toMatchSnapshot()
    })
  })
  describe('type returns', () => {
    it('appState', async () => {
      await store.setAppState('active')
      const storedValue = await store.getAppState()
      const storedType = typeof storedValue
      await expect(storedType).toMatchSnapshot()
      await expect(storedValue).toMatchSnapshot()
    })
    it('nodeOnline', async () => {
      await store.setNodeOnline(true)
      const storedValue = await store.getNodeOnline()
      const storedType = typeof storedValue
      await expect(storedType).toMatchSnapshot()
      await expect(storedValue).toMatchSnapshot()
    })
    it('nodeOnline', async () => {
      const newState = { state: NodeState.creating }
      await store.setNodeState(newState)
      const storedValue = await store.getNodeState()
      const storedType = typeof storedValue
      await expect(storedType).toMatchSnapshot()
      await expect(storedValue).toMatchSnapshot()
    })
  })
})
