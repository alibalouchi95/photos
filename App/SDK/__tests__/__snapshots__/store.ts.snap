// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rn store type returns appState 1`] = `undefined`;

exports[`rn store type returns appState 2`] = `undefined`;

exports[`rn store type returns appState 3`] = `undefined`;

exports[`rn store type returns appState 4`] = `"string"`;

exports[`rn store type returns appState 5`] = `"background"`;

exports[`rn store type returns nodeOnline 1`] = `"boolean"`;

exports[`rn store type returns nodeOnline 2`] = `true`;

exports[`rn store type returns nodeOnline 3`] = `"object"`;

exports[`rn store type returns nodeOnline 4`] = `
Object {
  "state": "creating",
}
`;
